2-projects/1-todo-list/script.js

```diff
@@ -17,6 +62,55 @@ function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  const liEl = document.createElement("li");
+
+  liEl.setAttribute(
+    "class",
+    "list-group-item d-flex justify-content-between align-items-center"
+  );
+
+  const inputValue = document.querySelector("#todoInput").value;
+  const textValueFromInput = document.createTextNode(inputValue);
+
+  liEl.appendChild(textValueFromInput);
+
+  if (inputValue === "") {
+    alert("It is empty, write something.");
+  } else {
+    ulEl.appendChild(liEl);
+  }
+
+  document.querySelector("#todoInput").value = "";
```

Comment:
1. Very good, I see you tried to prevent empty todo item creation. You need to explicitly call **return** after alert otherwise all code after line 78 will run and an empty todo item will be added.
2. Since we will stop once we find the input field is empty. So, lets get the input value and check it before all other actions to avoid unnecessary actions like creating new elements.
3.